File: facts.d/get_extensions_details.php
#!/usr/local/zend/bin/php
<?php
error_reporting(E_ERROR);

$zs_api_config_file='/.zsapi.ini';
$zs_api_target='localadmin';
$zs_api_config=parse_ini_file($zs_api_config_file, true);

$api_key_name=$zs_api_config[$zs_api_target]['zskey'];
$api_key_hash=$zs_api_config[$zs_api_target]['zssecret'];

$get_extensions_details="/usr/local/zend/bin/zs-client.sh configurationExtensionsList --target=localadmin --output-format=json > /tmp/ZendExtensionsList";

exec($get_extensions_details , $extensions_exec_output);

$textJson = file_get_contents('/tmp/ZendExtensionsList');

//die ($extensions_exec_output);

//print_r ($extensions_exec_output);

//die;

$arr = json_decode($textJson, true);

//print_r ($arr);
//var_dump ($arr['extension']);
//print_r ($extensions_exec_output[2]);

foreach($arr['responseData']['extensions'] as $app) {
    $name = $app['name'];

    foreach($app as $key => $value) {
        if(! is_array($value))
            echo 'zend_extension_' . $key . '_' . $name . '=' . $value . PHP_EOL;
    }
}

File: facts.d/get_extensions_details.py
#!/usr/bin/python

import json
import os.path
import re
import subprocess

zs_api_config_file = '/.zsapi.ini'
zs_api_target = 'localadmin'

if os.path.isfile("/usr/local/zend/bin/zs-client.sh"):

  extensions_details = subprocess.check_output(["/usr/local/zend/bin/zs-client.sh", "configurationExtensionsList", "--target=localadmin", "--output-format=json"])

  ## Strip the PHP notices from the json
  extensions_details = re.sub("Notice:.*\n", "", extensions_details)
  ## Strip the newlines from the json
  extensions_details = re.sub("\n", "", extensions_details)

  arr = json.loads(extensions_details)

  for extension in arr[u"responseData"][u"extensions"]:
    name = extension["name"]
    for key, value in extension.iteritems():
      if not isinstance(value, list):
        print ('zend_extension_' + key + '_' + name + '=' + value)
